// neobrutal/dist/index.js
#!/usr/bin/env node

// src/index.ts
import { Command as Command6 } from "commander";
import { createRequire } from "module";

// src/commands/init.ts
import path3 from "path";
import fs3 from "fs-extra";
import { Command } from "commander";
import prompts from "prompts";
import { z as z2 } from "zod";

// src/utils/logger.ts
import chalk from "chalk";
var logger = {
  log: (...args) => {
    console.log(...args);
  },
  info: (...args) => {
    console.log(chalk.blue("info"), ...args);
  },
  success: (...args) => {
    console.log(chalk.green("\u2714"), ...args);
  },
  warn: (...args) => {
    console.log(chalk.yellow("warning"), ...args);
  },
  error: (...args) => {
    console.log(chalk.red("error"), ...args);
  },
  break: () => {
    console.log("");
  }
};
var highlighter = {
  info: (text) => chalk.cyan(text),
  success: (text) => chalk.green(text),
  warn: (text) => chalk.yellow(text),
  error: (text) => chalk.red(text),
  code: (text) => chalk.gray(text)
};

// src/utils/spinner.ts
import ora from "ora";
function spinner(text) {
  return ora({
    text,
    spinner: "dots"
  });
}

// src/utils/errors.ts
function handleError(error) {
  if (typeof error === "string") {
    logger.error(error);
    process.exit(1);
  }
  if (error instanceof Error) {
    logger.error(error.message);
    process.exit(1);
  }
  logger.error("Something went wrong. Please try again.");
  process.exit(1);
}

// src/utils/config.ts
import path from "path";
import fs from "fs-extra";
import { z } from "zod";
var DEFAULT_REGISTRY_URL = "https://www.neobrutalui.live/r";
function resolveRegistryUrl() {
  const envUrl = process.env.NEOBRUTAL_REGISTRY_URL?.trim();
  if (!envUrl) {
    return DEFAULT_REGISTRY_URL;
  }
  return envUrl.replace(/\/+$/, "");
}
var REGISTRY_URL = resolveRegistryUrl();
var configSchema = z.object({
  $schema: z.string().optional(),
  style: z.string().default("default"),
  rsc: z.boolean().default(true),
  tsx: z.boolean().default(true),
  tailwind: z.object({
    config: z.string().optional(),
    css: z.string(),
    baseColor: z.string().default("neutral"),
    cssVariables: z.boolean().default(true),
    prefix: z.string().optional()
  }),
  aliases: z.object({
    components: z.string(),
    utils: z.string(),
    ui: z.string().optional(),
    lib: z.string().optional(),
    hooks: z.string().optional()
  })
});
async function getConfig(cwd) {
  const configPath = path.resolve(cwd, "components.json");
  if (!await fs.pathExists(configPath)) {
    return null;
  }
  try {
    const configContent = await fs.readFile(configPath, "utf-8");
    const rawConfig = JSON.parse(configContent);
    return configSchema.parse(rawConfig);
  } catch {
    return null;
  }
}
async function writeConfig(cwd, config) {
  const configPath = path.resolve(cwd, "components.json");
  await fs.writeFile(
    configPath,
    JSON.stringify(config, null, 2) + "\n",
    "utf-8"
  );
}
async function getProjectInfo(cwd) {
  const [packageJson2, srcDir, typescript, tailwindCss, aliasPrefix, cssPath] = await Promise.all([
    getPackageJson(cwd),
    fs.pathExists(path.resolve(cwd, "src")),
    fs.pathExists(path.resolve(cwd, "tsconfig.json")),
    detectTailwindCss(cwd),
    getTsConfigAliasPrefix(cwd),
    detectCssPath(cwd)
  ]);
  if (!packageJson2) {
    return {
      framework: "unknown",
      srcDir,
      typescript,
      tailwindCss,
      aliasPrefix,
      cssPath
    };
  }
  const deps = {
    ...packageJson2.dependencies ?? {},
    ...packageJson2.devDependencies ?? {}
  };
  if (deps.next) {
    const hasAppDir = await fs.pathExists(path.resolve(cwd, "app")) || await fs.pathExists(path.resolve(cwd, "src/app"));
    return {
      framework: hasAppDir ? "next-app" : "next-pages",
      srcDir,
      typescript,
      tailwindCss,
      aliasPrefix,
      cssPath
    };
  }
  if (deps["@remix-run/react"]) {
    return {
      framework: "remix",
      srcDir,
      typescript,
      tailwindCss,
      aliasPrefix,
      cssPath
    };
  }
  if (deps.astro) {
    return {
      framework: "astro",
      srcDir,
      typescript,
      tailwindCss,
      aliasPrefix,
      cssPath
    };
  }
  if (deps.vite) {
    return {
      framework: "vite",
      srcDir,
      typescript,
      tailwindCss,
      aliasPrefix,
      cssPath
    };
  }
  return {
    framework: "unknown",
    srcDir,
    typescript,
    tailwindCss,
    aliasPrefix,
    cssPath
  };
}
async function getPackageJson(cwd) {
  const packageJsonPath = path.resolve(cwd, "package.json");
  if (!await fs.pathExists(packageJsonPath)) {
    return null;
  }
  return fs.readJson(packageJsonPath);
}
async function detectTailwindCss(cwd) {
  const packageJson2 = await getPackageJson(cwd);
  if (!packageJson2) {
    return false;
  }
  const deps = {
    ...packageJson2.dependencies ?? {},
    ...packageJson2.devDependencies ?? {}
  };
  return !!deps.tailwindcss;
}
async function getTsConfigAliasPrefix(cwd) {
  const tsconfigPath = path.resolve(cwd, "tsconfig.json");
  if (!await fs.pathExists(tsconfigPath)) {
    return "@/";
  }
  try {
    const content = await fs.readFile(tsconfigPath, "utf-8");
    const jsonContent = content.replace(/\/\*[\s\S]*?\*\/|\/\/.*/g, "");
    const tsconfig = JSON.parse(jsonContent);
    const paths = tsconfig.compilerOptions?.paths;
    if (!paths) {
      return "@/";
    }
    for (const [alias, targets] of Object.entries(paths)) {
      if (alias.endsWith("/*") && Array.isArray(targets)) {
        const prefix = alias.replace("/*", "/");
        if (["@/", "~/", "#/", "$/"].includes(prefix)) {
          return prefix;
        }
      }
    }
    const firstAlias = Object.keys(paths)[0];
    if (firstAlias?.endsWith("/*")) {
      return firstAlias.replace("/*", "/");
    }
    return "@/";
  } catch {
    return "@/";
  }
}
async function detectCssPath(cwd) {
  const possiblePaths = [
    "app/globals.css",
    "src/app/globals.css",
    "src/index.css",
    "src/styles/globals.css",
    "styles/globals.css",
    "app/global.css",
    "src/app/global.css"
  ];
  for (const cssPath of possiblePaths) {
    if (await fs.pathExists(path.resolve(cwd, cssPath))) {
      return cssPath;
    }
  }
  return null;
}

// src/utils/package-manager.ts
import path2 from "path";
import fs2 from "fs-extra";
import { spawn } from "child_process";
var PACKAGE_MANAGERS = {
  pnpm: {
    name: "pnpm",
    installCommand: "pnpm",
    addFlag: "add",
    devFlag: "-D",
    lockfile: "pnpm-lock.yaml"
  },
  yarn: {
    name: "yarn",
    installCommand: "yarn",
    addFlag: "add",
    devFlag: "-D",
    lockfile: "yarn.lock"
  },
  bun: {
    name: "bun",
    installCommand: "bun",
    addFlag: "add",
    devFlag: "-d",
    lockfile: "bun.lock"
  },
  npm: {
    name: "npm",
    installCommand: "npm",
    addFlag: "install",
    devFlag: "--save-dev",
    lockfile: "package-lock.json"
  }
};
function getInstallProcess(packageManager) {
  const config = PACKAGE_MANAGERS[packageManager];
  if (process.platform !== "win32") {
    return { command: config.installCommand, shell: false };
  }
  const windowsCommandMap = {
    npm: "npm.cmd",
    pnpm: "pnpm.cmd",
    yarn: "yarn.cmd",
    bun: "bun.exe"
  };
  return {
    command: windowsCommandMap[packageManager] ?? config.installCommand,
    shell: false
  };
}
async function detectPackageManagerFromPackageJson(cwd) {
  const packageJsonPath = path2.resolve(cwd, "package.json");
  if (!await fs2.pathExists(packageJsonPath)) {
    return null;
  }
  try {
    const content = await fs2.readFile(packageJsonPath, "utf-8");
    const pkg = JSON.parse(content);
    const packageManager = pkg.packageManager?.trim().toLowerCase() ?? "";
    if (packageManager.startsWith("pnpm@") || packageManager === "pnpm") {
      return "pnpm";
    }
    if (packageManager.startsWith("yarn@") || packageManager === "yarn") {
      return "yarn";
    }
    if (packageManager.startsWith("bun@") || packageManager === "bun") {
      return "bun";
    }
    if (packageManager.startsWith("npm@") || packageManager === "npm") {
      return "npm";
    }
  } catch {
    return null;
  }
  return null;
}
async function detectPackageManager(cwd) {
  const packageManagerFromConfig = await detectPackageManagerFromPackageJson(cwd);
  if (packageManagerFromConfig) {
    return packageManagerFromConfig;
  }
  const checks = [
    { pm: "pnpm", lockfile: "pnpm-lock.yaml" },
    { pm: "yarn", lockfile: "yarn.lock" },
    { pm: "bun", lockfile: "bun.lock" },
    { pm: "bun", lockfile: "bun.lockb" },
    { pm: "npm", lockfile: "package-lock.json" }
  ];
  for (const { pm, lockfile } of checks) {
    if (await fs2.pathExists(path2.resolve(cwd, lockfile))) {
      return pm;
    }
  }
  return "npm";
}
async function getInstalledDependencies(cwd) {
  const packageJsonPath = path2.resolve(cwd, "package.json");
  const installed = /* @__PURE__ */ new Set();
  try {
    if (await fs2.pathExists(packageJsonPath)) {
      const content = await fs2.readFile(packageJsonPath, "utf-8");
      const pkg = JSON.parse(content);
      if (pkg.dependencies) {
        Object.keys(pkg.dependencies).forEach((dep) => installed.add(dep));
      }
      if (pkg.devDependencies) {
        Object.keys(pkg.devDependencies).forEach((dep) => installed.add(dep));
      }
    }
  } catch {
  }
  return installed;
}
function getInstallCommand(packageManager, dependencies, isDev = false) {
  const config = PACKAGE_MANAGERS[packageManager];
  const devFlag = isDev ? ` ${config.devFlag}` : "";
  return `${config.installCommand} ${config.addFlag}${devFlag} ${dependencies.join(" ")}`;
}
async function installDependencies(cwd, dependencies, options = {}) {
  const { isDev = false, silent = false } = options;
  if (dependencies.length === 0) {
    return true;
  }
  const packageManager = await detectPackageManager(cwd);
  const config = PACKAGE_MANAGERS[packageManager];
  const installProcess = getInstallProcess(packageManager);
  const args = [config.addFlag];
  if (isDev) {
    args.push(config.devFlag);
  }
  args.push(...dependencies);
  if (!silent) {
    logger.info(`Installing dependencies with ${packageManager}...`);
  }
  return new Promise((resolve) => {
    const child = spawn(installProcess.command, args, {
      cwd,
      stdio: silent ? "ignore" : "inherit",
      shell: installProcess.shell
    });
    child.on("close", (code) => {
      resolve(code === 0);
    });
    child.on("error", () => {
      resolve(false);
    });
  });
}

// src/commands/init.ts
var initOptionsSchema = z2.object({
  cwd: z2.string(),
  yes: z2.boolean(),
  force: z2.boolean(),
  skipCss: z2.boolean()
});
var init = new Command().name("init").description("initialize your project and install dependencies").option(
  "-c, --cwd <cwd>",
  "the working directory. defaults to the current directory.",
  process.cwd()
).option("-y, --yes", "skip confirmation prompt.", false).option("-f, --force", "force overwrite of existing configuration.", false).option("--skip-css", "skip CSS variable injection.", false).action(async (opts) => {
  try {
    const options = initOptionsSchema.parse({
      cwd: path3.resolve(opts.cwd),
      yes: opts.yes,
      force: opts.force,
      skipCss: opts.skipCss
    });
    await runInit(options);
  } catch (error) {
    handleError(error);
  }
});
async function runInit(options) {
  const { cwd, yes, force, skipCss } = options;
  const existingConfig = await getConfig(cwd);
  if (existingConfig && !force) {
    logger.warn(
      `A ${highlighter.info("components.json")} file already exists at ${highlighter.info(cwd)}.`
    );
    const { overwrite } = await prompts({
      type: "confirm",
      name: "overwrite",
      message: "Would you like to overwrite it?",
      initial: false
    });
    if (!overwrite) {
      logger.info("Initialization cancelled.");
      return;
    }
  }
  const projectInfo = await getProjectInfo(cwd);
  const frameworkDisplay = getFrameworkDisplayName(projectInfo.framework);
  logger.info(`Detected ${highlighter.info(frameworkDisplay)} project.`);
  if (projectInfo.tailwindCss) {
    logger.info(`Tailwind CSS detected.`);
  }
  if (projectInfo.aliasPrefix !== "@/") {
    logger.info(`Using alias prefix: ${highlighter.info(projectInfo.aliasPrefix)}`);
  }
  logger.break();
  let config;
  if (yes) {
    config = getDefaultConfig(projectInfo);
  } else {
    config = await promptForConfig(projectInfo);
  }
  const initSpinner = spinner("Initializing project...").start();
  await ensureDirectories(cwd, config);
  await createUtilsFile(cwd, config);
  await writeConfig(cwd, config);
  initSpinner.succeed("Project initialized successfully.");
  if (!skipCss && config.tailwind?.css) {
    const cssPath = path3.resolve(cwd, config.tailwind.css);
    if (await fs3.pathExists(cssPath)) {
      const cssSpinner = spinner("Injecting CSS variables...").start();
      const injected = await injectCssVariables(cssPath);
      if (injected) {
        cssSpinner.succeed("Neobrutalism CSS variables added.");
      } else {
        cssSpinner.succeed("CSS variables already present.");
      }
    }
  }
  const baseDeps = ["clsx", "tailwind-merge"];
  const packageManager = await detectPackageManager(cwd);
  logger.break();
  logger.info("Installing base dependencies...");
  const depsSpinner = spinner(
    `Running ${getInstallCommand(packageManager, baseDeps)}`
  ).start();
  const success = await installDependencies(cwd, baseDeps, { silent: true });
  if (success) {
    depsSpinner.succeed("Base dependencies installed.");
  } else {
    depsSpinner.fail("Failed to install dependencies.");
    logger.break();
    logger.warn("Please install manually:");
    logger.log(`  ${getInstallCommand(packageManager, baseDeps)}`);
  }
  logger.break();
  logger.success(
    `${highlighter.success("Success!")} Project initialization completed.`
  );
  logger.info("You may now add components using:");
  logger.break();
  logger.log(`  ${highlighter.code("npx neobrutal add button")}`);
  logger.break();
}
function getFrameworkDisplayName(framework) {
  const displayNames = {
    "next-app": "Next.js (App Router)",
    "next-pages": "Next.js (Pages Router)",
    vite: "Vite",
    remix: "Remix",
    astro: "Astro",
    unknown: "Unknown"
  };
  return displayNames[framework] || framework;
}
function getDefaultConfig(projectInfo) {
  const baseDir = projectInfo.srcDir ? "src/" : "";
  const prefix = projectInfo.aliasPrefix;
  return {
    style: "default",
    rsc: projectInfo.framework === "next-app",
    tsx: projectInfo.typescript,
    tailwind: {
      config: "",
      css: projectInfo.cssPath || `${baseDir}app/globals.css`,
      baseColor: "neutral",
      cssVariables: true,
      prefix: ""
    },
    aliases: {
      components: `${prefix}components`,
      utils: `${prefix}lib/utils`,
      ui: `${prefix}components/ui`,
      lib: `${prefix}lib`,
      hooks: `${prefix}hooks`
    }
  };
}
async function promptForConfig(projectInfo) {
  const baseDir = projectInfo.srcDir ? "src/" : "";
  const prefix = projectInfo.aliasPrefix;
  const { cssPath } = await prompts({
    type: "text",
    name: "cssPath",
    message: "Where is your global CSS file?",
    initial: projectInfo.cssPath || `${baseDir}app/globals.css`
  });
  const { componentsAlias } = await prompts({
    type: "text",
    name: "componentsAlias",
    message: "Configure the import alias for components:",
    initial: `${prefix}components`
  });
  const { utilsAlias } = await prompts({
    type: "text",
    name: "utilsAlias",
    message: "Configure the import alias for utils:",
    initial: `${prefix}lib/utils`
  });
  return {
    style: "default",
    rsc: projectInfo.framework === "next-app",
    tsx: projectInfo.typescript,
    tailwind: {
      config: "",
      css: cssPath,
      baseColor: "neutral",
      cssVariables: true,
      prefix: ""
    },
    aliases: {
      components: componentsAlias,
      utils: utilsAlias,
      ui: `${componentsAlias}/ui`,
      lib: `${prefix}lib`,
      hooks: `${prefix}hooks`
    }
  };
}
function stripAliasPrefix(aliasPath) {
  const prefixes = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixes) {
    if (aliasPath.startsWith(prefix)) {
      return aliasPath.slice(prefix.length);
    }
  }
  return aliasPath;
}
async function ensureDirectories(cwd, config) {
  const dirs = [
    stripAliasPrefix(config.aliases.components),
    stripAliasPrefix(config.aliases.ui || "components/ui"),
    stripAliasPrefix(config.aliases.lib || "lib")
  ];
  for (const dir of dirs) {
    await fs3.ensureDir(path3.resolve(cwd, dir));
  }
}
async function createUtilsFile(cwd, config) {
  const utilsPath = path3.resolve(
    cwd,
    stripAliasPrefix(config.aliases.utils) + ".ts"
  );
  if (await fs3.pathExists(utilsPath)) {
    return;
  }
  const utilsContent = `import { clsx, type ClassValue } from "clsx"
import { twMerge } from "tailwind-merge"

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs))
}
`;
  await fs3.ensureDir(path3.dirname(utilsPath));
  await fs3.writeFile(utilsPath, utilsContent, "utf-8");
}
var NEOBRUTAL_CSS_VARIABLES = `
@theme inline {
  --color-white: var(--white);
  --color-black: var(--black);
  --color-bg: var(--bg);
  --color-main: var(--main);
  --color-mint: var(--mint);
  --color-lemon: var(--lemon);
  --color-hot-pink: var(--hot-pink);
  --shadow-brutal: 4px 4px 0px 0px var(--black);
  --radius-base: var(--radius);
}

:root {
  --white: #ffffff;
  --black: #000000;
  --main: #b6ace4;
  --bg: #f0eefc;
  --radius: 5px;
  --mint: #97ee88;
  --lemon: #fed170;
  --hot-pink: #ee8888;
}

@layer utilities {
  .shadow-brutal {
    box-shadow: var(--shadow-brutal);
  }
}
`;
async function injectCssVariables(cssPath) {
  const content = await fs3.readFile(cssPath, "utf-8");
  if (content.includes("--main:") && content.includes("--shadow-brutal:")) {
    return false;
  }
  let newContent;
  if (content.includes("@import")) {
    const importEndIndex = content.lastIndexOf("@import");
    const lineEnd = content.indexOf(";", importEndIndex);
    newContent = content.slice(0, lineEnd + 1) + "\n" + NEOBRUTAL_CSS_VARIABLES + content.slice(lineEnd + 1);
  } else {
    newContent = NEOBRUTAL_CSS_VARIABLES + "\n" + content;
  }
  await fs3.writeFile(cssPath, newContent, "utf-8");
  return true;
}

// src/commands/add.ts
import path4 from "path";
import fs4 from "fs-extra";
import { Command as Command2 } from "commander";
import prompts2 from "prompts";
import { z as z4 } from "zod";

// src/utils/registry.ts
import { z as z3 } from "zod";
var MAX_RETRIES = 3;
var RETRY_DELAY_MS = 1e3;
var REQUEST_TIMEOUT_MS = 1e4;
var COMPONENT_NAME_PATTERN = /^[a-z0-9][a-z0-9-]*$/;
var SAFE_FILE_PATH_PATTERN = /^(?!\.)(?!.*(?:^|\/)\.\.(?:\/|$))[a-zA-Z0-9._/-]+$/;
var registryIndexCache = null;
var registryItemCache = /* @__PURE__ */ new Map();
function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}
function normalizeComponentName(name) {
  return name.trim().toLowerCase();
}
function isSafeRegistryFilePath(filePath) {
  const normalized = filePath.replace(/\\/g, "/");
  return SAFE_FILE_PATH_PATTERN.test(normalized);
}
var registryItemFileSchema = z3.object({
  path: z3.string().refine(isSafeRegistryFilePath, {
    message: "Invalid registry file path."
  }),
  content: z3.string(),
  type: z3.enum(["registry:ui", "registry:lib", "registry:hook", "registry:component"])
});
var registryItemSchema = z3.object({
  name: z3.string().regex(COMPONENT_NAME_PATTERN, "Invalid registry item name."),
  type: z3.enum(["registry:ui", "registry:lib", "registry:hook", "registry:component", "registry:style"]),
  description: z3.string().optional(),
  dependencies: z3.array(z3.string()).optional(),
  devDependencies: z3.array(z3.string()).optional(),
  registryDependencies: z3.array(z3.string()).optional(),
  files: z3.array(registryItemFileSchema),
  tailwind: z3.object({
    config: z3.record(z3.unknown()).optional()
  }).optional(),
  cssVars: z3.record(z3.record(z3.string())).optional(),
  docs: z3.string().optional()
});
var registryIndexSchema = z3.array(
  z3.object({
    name: z3.string().regex(COMPONENT_NAME_PATTERN, "Invalid registry item name."),
    type: z3.enum(["registry:ui", "registry:lib", "registry:hook", "registry:component", "registry:style"]),
    description: z3.string().optional(),
    dependencies: z3.array(z3.string()).optional(),
    registryDependencies: z3.array(z3.string()).optional()
  })
);
async function fetchWithRetry(url, retries = MAX_RETRIES) {
  let lastError;
  for (let attempt = 0; attempt < retries; attempt++) {
    const controller = new AbortController();
    const timeout = setTimeout(() => controller.abort(), REQUEST_TIMEOUT_MS);
    try {
      const response = await fetch(url, {
        signal: controller.signal,
        headers: {
          Accept: "application/json"
        }
      });
      if (response.ok) {
        clearTimeout(timeout);
        return response;
      }
      if (response.status === 404) {
        throw new Error(`Not found: ${url}`);
      }
      if (response.status >= 400 && response.status < 500 && response.status !== 429) {
        throw new Error(`Request failed: HTTP ${response.status} ${response.statusText}`);
      }
      lastError = new Error(`HTTP ${response.status}: ${response.statusText}`);
    } catch (error) {
      if (error instanceof Error && error.name === "AbortError") {
        lastError = new Error(`Request timed out after ${REQUEST_TIMEOUT_MS}ms: ${url}`);
      } else {
        lastError = error instanceof Error ? error : new Error(String(error));
      }
    } finally {
      clearTimeout(timeout);
    }
    if (attempt < retries - 1) {
      await sleep(RETRY_DELAY_MS * (attempt + 1));
    }
  }
  throw lastError ?? new Error(`Failed to fetch ${url}`);
}
async function fetchJson(url) {
  const response = await fetchWithRetry(url);
  try {
    return await response.json();
  } catch {
    throw new Error(`Invalid JSON response from ${url}`);
  }
}
async function getRegistryIndex() {
  if (registryIndexCache) {
    return registryIndexCache;
  }
  const json = await fetchJson(`${REGISTRY_URL}/index.json`);
  const parsed = registryIndexSchema.parse(json);
  registryIndexCache = parsed;
  return parsed;
}
async function getRegistryItem(name) {
  const normalizedName = normalizeComponentName(name);
  if (!COMPONENT_NAME_PATTERN.test(normalizedName)) {
    throw new Error(`Invalid component name: ${name}`);
  }
  const cached = registryItemCache.get(normalizedName);
  if (cached) {
    return cached;
  }
  const json = await fetchJson(`${REGISTRY_URL}/${normalizedName}.json`);
  const parsed = registryItemSchema.parse(json);
  registryItemCache.set(normalizedName, parsed);
  return parsed;
}
async function getRegistryItems(names) {
  const uniqueNames = Array.from(
    new Set(names.map((name) => normalizeComponentName(name)).filter(Boolean))
  );
  const results = await Promise.all(uniqueNames.map((name) => getRegistryItem(name)));
  return results;
}
async function resolveRegistryDependencies(items) {
  const resolved = /* @__PURE__ */ new Map();
  const inProgress = /* @__PURE__ */ new Set();
  async function visit(item, stack) {
    if (resolved.has(item.name)) {
      return;
    }
    if (inProgress.has(item.name)) {
      throw new Error(
        `Circular registry dependency detected: ${[...stack, item.name].join(" -> ")}`
      );
    }
    inProgress.add(item.name);
    if (item.registryDependencies?.length) {
      for (const dependencyName of item.registryDependencies) {
        const depItem = await getRegistryItem(dependencyName);
        await visit(depItem, [...stack, item.name]);
      }
    }
    inProgress.delete(item.name);
    resolved.set(item.name, item);
  }
  for (const item of items) {
    await visit(item, []);
  }
  return Array.from(resolved.values());
}

// src/utils/transform.ts
function transformImports(content, config) {
  let transformed = content;
  const utilsPrefix = extractAliasPrefix(config.aliases.utils);
  const componentsPrefix = extractAliasPrefix(config.aliases.components);
  if (utilsPrefix === "@/" && componentsPrefix === "@/") {
    return transformed;
  }
  if (utilsPrefix !== "@/") {
    const utilsPath = config.aliases.utils;
    transformed = transformed.replace(
      /from\s+["']@\/lib\/utils["']/g,
      `from "${utilsPath}"`
    );
  }
  if (componentsPrefix !== "@/") {
    const componentsBase = config.aliases.components;
    transformed = transformed.replace(
      /from\s+["']@\/components\/(.*?)["']/g,
      `from "${componentsBase}/$1"`
    );
  }
  if (utilsPrefix !== "@/") {
    transformed = transformed.replace(
      /from\s+["']@\/(.*?)["']/g,
      `from "${utilsPrefix}$1"`
    );
  }
  return transformed;
}
function extractAliasPrefix(alias) {
  const prefixPatterns = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixPatterns) {
    if (alias.startsWith(prefix)) {
      return prefix;
    }
  }
  const slashIndex = alias.indexOf("/");
  if (slashIndex > 0) {
    return alias.substring(0, slashIndex + 1);
  }
  return alias;
}

// src/commands/add.ts
var addOptionsSchema = z4.object({
  components: z4.array(z4.string()).optional(),
  cwd: z4.string(),
  yes: z4.boolean(),
  overwrite: z4.boolean(),
  all: z4.boolean()
});
var add = new Command2().name("add").description("add components to your project").argument("[components...]", "the components to add").option(
  "-c, --cwd <cwd>",
  "the working directory. defaults to the current directory.",
  process.cwd()
).option("-y, --yes", "skip confirmation prompt.", false).option("-o, --overwrite", "overwrite existing files.", false).option("-a, --all", "add all available components.", false).action(async (components, opts) => {
  try {
    const options = addOptionsSchema.parse({
      components,
      cwd: path4.resolve(opts.cwd),
      yes: opts.yes,
      overwrite: opts.overwrite,
      all: opts.all
    });
    await runAdd(options);
  } catch (error) {
    handleError(error);
  }
});
async function runAdd(options) {
  const { cwd, overwrite, all } = options;
  let { components } = options;
  let config = await getConfig(cwd);
  if (!config) {
    logger.warn(
      `No ${highlighter.info("components.json")} file found at ${highlighter.info(cwd)}.`
    );
    const { proceed } = await prompts2({
      type: "confirm",
      name: "proceed",
      message: "Would you like to initialize the project first?",
      initial: true
    });
    if (!proceed) {
      logger.info("Add cancelled.");
      return;
    }
    await runInit({ cwd, yes: false, force: false, skipCss: false });
    config = await getConfig(cwd);
    if (!config) {
      logger.error("Failed to initialize project.");
      return;
    }
  }
  const registryIndex = await getRegistryIndex();
  if (all) {
    components = registryIndex.filter((item) => item.type === "registry:ui").map((item) => item.name);
  }
  if (!components || components.length === 0) {
    const { selectedComponents } = await prompts2({
      type: "multiselect",
      name: "selectedComponents",
      message: "Which components would you like to add?",
      hint: "Space to select. A to toggle all. Enter to submit.",
      instructions: false,
      choices: registryIndex.filter((item) => item.type === "registry:ui").map((item) => ({
        title: item.name,
        value: item.name,
        description: item.description
      }))
    });
    if (!selectedComponents?.length) {
      logger.info("No components selected. Exiting.");
      return;
    }
    components = selectedComponents;
  }
  if (!components || components.length === 0) {
    logger.info("No components selected. Exiting.");
    return;
  }
  components = normalizeComponentNames(components);
  const availableComponents = new Set(registryIndex.map((item) => item.name));
  const invalidComponents = components.filter(
    (component) => !availableComponents.has(component)
  );
  if (invalidComponents.length > 0) {
    logger.error(
      `The following components are not available: ${invalidComponents.join(", ")}`
    );
    logger.info("Run `npx neobrutal list` to see all available components.");
    return;
  }
  const addSpinner = spinner("Fetching components...").start();
  try {
    const items = await getRegistryItems(components);
    const resolvedItems = await resolveRegistryDependencies(items);
    addSpinner.succeed(`Found ${resolvedItems.length} component(s).`);
    const npmDependencies = /* @__PURE__ */ new Set();
    const npmDevDependencies = /* @__PURE__ */ new Set();
    const filesToWrite = [];
    for (const item of resolvedItems) {
      if (item.dependencies) {
        item.dependencies.forEach((dep) => npmDependencies.add(dep));
      }
      if (item.devDependencies) {
        item.devDependencies.forEach((dep) => npmDevDependencies.add(dep));
      }
      for (const file of item.files) {
        const targetPath = resolveFilePath(cwd, config, file.path);
        if (await fs4.pathExists(targetPath)) {
          if (file.path === "lib/utils.ts") {
            const existingContent = await fs4.readFile(targetPath, "utf-8");
            const transformedContent2 = transformImports(file.content, config);
            const normalizedExisting = existingContent.replace(/\r\n/g, "\n").trim();
            const normalizedNew = transformedContent2.replace(/\r\n/g, "\n").trim();
            if (normalizedExisting === normalizedNew) {
              continue;
            }
          }
          if (!overwrite) {
            const { shouldOverwrite } = await prompts2({
              type: "confirm",
              name: "shouldOverwrite",
              message: `File ${highlighter.info(path4.relative(cwd, targetPath))} already exists. Overwrite?`,
              initial: false
            });
            if (!shouldOverwrite) {
              logger.info(`Skipping ${file.path}`);
              continue;
            }
          }
        }
        const transformedContent = transformImports(file.content, config);
        filesToWrite.push({
          path: targetPath,
          content: transformedContent
        });
      }
    }
    const writeSpinner = spinner("Writing files...").start();
    for (const file of filesToWrite) {
      await fs4.ensureDir(path4.dirname(file.path));
      await fs4.writeFile(file.path, file.content, "utf-8");
    }
    writeSpinner.succeed(`Wrote ${filesToWrite.length} file(s).`);
    if (npmDependencies.size > 0 || npmDevDependencies.size > 0) {
      const installedDeps = await getInstalledDependencies(cwd);
      const depsToInstall = Array.from(npmDependencies).filter(
        (dep) => !installedDeps.has(dep)
      );
      const devDepsToInstall = Array.from(npmDevDependencies).filter(
        (dep) => !installedDeps.has(dep) && !depsToInstall.includes(dep)
      );
      const totalDepsToInstall = depsToInstall.length + devDepsToInstall.length;
      if (totalDepsToInstall > 0) {
        const packageManager = await detectPackageManager(cwd);
        logger.break();
        logger.info(`Installing ${totalDepsToInstall} dependencies...`);
        if (depsToInstall.length > 0) {
          const installSpinner = spinner(
            `Running ${getInstallCommand(packageManager, depsToInstall)}`
          ).start();
          const success = await installDependencies(cwd, depsToInstall, {
            silent: true
          });
          if (success) {
            installSpinner.succeed("Dependencies installed.");
          } else {
            installSpinner.fail("Failed to install dependencies.");
            logger.break();
            logger.warn("Please install manually:");
            logger.log(`  ${getInstallCommand(packageManager, depsToInstall)}`);
          }
        }
        if (devDepsToInstall.length > 0) {
          const installDevSpinner = spinner(
            `Running ${getInstallCommand(packageManager, devDepsToInstall, true)}`
          ).start();
          const success = await installDependencies(cwd, devDepsToInstall, {
            isDev: true,
            silent: true
          });
          if (success) {
            installDevSpinner.succeed("Dev dependencies installed.");
          } else {
            installDevSpinner.fail("Failed to install dev dependencies.");
            logger.break();
            logger.warn("Please install manually:");
            logger.log(`  ${getInstallCommand(packageManager, devDepsToInstall, true)}`);
          }
        }
      }
    }
    logger.break();
    logger.success(`${highlighter.success("Success!")} Components added.`);
    logger.break();
    const addedComponents = resolvedItems.map((item) => item.name);
    logger.info(`Added: ${addedComponents.join(", ")}`);
  } catch (error) {
    addSpinner.fail("Failed to add components.");
    throw error;
  }
}
function normalizeComponentNames(components) {
  return Array.from(
    new Set(
      components.map((component) => component.trim().toLowerCase()).filter(Boolean)
    )
  );
}
function stripAliasPrefix2(aliasPath) {
  const prefixes = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixes) {
    if (aliasPath.startsWith(prefix)) {
      return aliasPath.slice(prefix.length);
    }
  }
  return aliasPath;
}
function ensurePathInProjectRoot(cwd, targetPath, sourcePath) {
  const relative = path4.relative(cwd, targetPath);
  if (relative.startsWith("..") || path4.isAbsolute(relative)) {
    throw new Error(`Refusing to write outside project root for ${sourcePath}`);
  }
  return targetPath;
}
function resolveFilePath(cwd, config, filePath) {
  const normalizedPath = filePath.replace(/\\/g, "/");
  if (normalizedPath.startsWith("/") || normalizedPath.includes("../")) {
    throw new Error(`Invalid registry file path: ${filePath}`);
  }
  if (filePath.startsWith("components/ui/")) {
    const uiAlias = config.aliases.ui || `${config.aliases.components}/ui`;
    const resolvedPath = stripAliasPrefix2(uiAlias);
    const targetPath2 = path4.resolve(
      cwd,
      filePath.replace("components/ui/", resolvedPath + "/")
    );
    return ensurePathInProjectRoot(cwd, targetPath2, filePath);
  }
  if (filePath.startsWith("lib/")) {
    const libAlias = config.aliases.lib || "@/lib";
    const resolvedPath = stripAliasPrefix2(libAlias);
    const targetPath2 = path4.resolve(
      cwd,
      filePath.replace("lib/", resolvedPath + "/")
    );
    return ensurePathInProjectRoot(cwd, targetPath2, filePath);
  }
  if (filePath.startsWith("hooks/")) {
    const hooksAlias = config.aliases.hooks || "@/hooks";
    const resolvedPath = stripAliasPrefix2(hooksAlias);
    const targetPath2 = path4.resolve(
      cwd,
      filePath.replace("hooks/", resolvedPath + "/")
    );
    return ensurePathInProjectRoot(cwd, targetPath2, filePath);
  }
  const targetPath = path4.resolve(cwd, filePath);
  return ensurePathInProjectRoot(cwd, targetPath, filePath);
}

// src/commands/list.ts
import { Command as Command3 } from "commander";
var list = new Command3().name("list").description("list all available components").action(async () => {
  try {
    await runList();
  } catch (error) {
    handleError(error);
  }
});
async function runList() {
  const listSpinner = spinner("Fetching components...").start();
  try {
    const registryIndex = await getRegistryIndex();
    listSpinner.succeed(`Found ${registryIndex.length} component(s).`);
    logger.break();
    const uiComponents = registryIndex.filter((item) => item.type === "registry:ui");
    const libComponents = registryIndex.filter((item) => item.type === "registry:lib");
    const hookComponents = registryIndex.filter((item) => item.type === "registry:hook");
    if (uiComponents.length > 0) {
      logger.log(highlighter.info("UI Components:"));
      logger.break();
      const maxNameLength = Math.max(...uiComponents.map((c) => c.name.length));
      for (const component of uiComponents) {
        const name = component.name.padEnd(maxNameLength + 2);
        const description = component.description || "";
        logger.log(`  ${highlighter.success(name)} ${description}`);
      }
      logger.break();
    }
    if (libComponents.length > 0) {
      logger.log(highlighter.info("Utilities:"));
      logger.break();
      for (const component of libComponents) {
        const description = component.description || "";
        logger.log(`  ${highlighter.success(component.name)} ${description}`);
      }
      logger.break();
    }
    if (hookComponents.length > 0) {
      logger.log(highlighter.info("Hooks:"));
      logger.break();
      for (const component of hookComponents) {
        const description = component.description || "";
        logger.log(`  ${highlighter.success(component.name)} ${description}`);
      }
      logger.break();
    }
    logger.info("To add a component, run:");
    logger.break();
    logger.log(`  ${highlighter.code("npx neobrutal add <component>")}`);
    logger.break();
  } catch (error) {
    listSpinner.fail("Failed to fetch components.");
    throw error;
  }
}

// src/commands/diff.ts
import path5 from "path";
import fs5 from "fs-extra";
import { Command as Command4 } from "commander";
import { z as z5 } from "zod";
import * as Diff from "diff";
var diffOptionsSchema = z5.object({
  component: z5.string(),
  cwd: z5.string(),
  noDiff: z5.boolean(),
  context: z5.number()
});
var diff = new Command4().name("diff").description("show differences between local and registry version").argument("<component>", "the component to diff").option(
  "-c, --cwd <cwd>",
  "the working directory. defaults to the current directory.",
  process.cwd()
).option(
  "--no-diff",
  "only show if files differ, don't show the actual diff",
  false
).option(
  "--context <lines>",
  "number of context lines to show",
  "3"
).action(async (component, opts) => {
  try {
    const options = diffOptionsSchema.parse({
      component,
      cwd: path5.resolve(opts.cwd),
      noDiff: opts.noDiff ?? false,
      context: parseInt(opts.context ?? "3", 10)
    });
    await runDiff(options);
  } catch (error) {
    handleError(error);
  }
});
async function runDiff(options) {
  const { component, cwd, noDiff, context } = options;
  const config = await getConfig(cwd);
  if (!config) {
    logger.error(
      `No ${highlighter.info("components.json")} file found at ${highlighter.info(cwd)}.`
    );
    logger.info("Run `npx neobrutal init` to initialize the project.");
    return;
  }
  const diffSpinner = spinner(`Checking ${component}...`).start();
  try {
    const registryItem = await getRegistryItem(component);
    diffSpinner.succeed(`Found ${component} in registry.`);
    let hasChanges = false;
    for (const file of registryItem.files) {
      const localPath = resolveFilePath2(cwd, config, file.path);
      if (!await fs5.pathExists(localPath)) {
        logger.break();
        logger.warn(`${highlighter.warn(file.path)} does not exist locally.`);
        logger.info("  Run `npx neobrutal add " + component + "` to add it.");
        hasChanges = true;
        continue;
      }
      const localContent = await fs5.readFile(localPath, "utf-8");
      const registryContent = transformImports(file.content, config);
      if (localContent === registryContent) {
        logger.break();
        logger.success(`${highlighter.success(file.path)} is up to date.`);
      } else {
        hasChanges = true;
        logger.break();
        logger.warn(`${highlighter.warn(file.path)} has local modifications.`);
        if (!noDiff) {
          printUnifiedDiff(file.path, localContent, registryContent, context);
        }
      }
    }
    if (hasChanges) {
      logger.break();
      logger.info("To update to the latest version, run:");
      logger.break();
      logger.log(`  ${highlighter.code(`npx neobrutal add ${component} --overwrite`)}`);
      logger.break();
    }
  } catch (error) {
    diffSpinner.fail(`Failed to check ${component}.`);
    throw error;
  }
}
function printUnifiedDiff(filePath, localContent, registryContent, contextLines) {
  const patch = Diff.createPatch(
    filePath,
    localContent,
    registryContent,
    "local",
    "registry",
    { context: contextLines }
  );
  const lines = patch.split("\n");
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    if (i < 2) {
      continue;
    }
    if (line.startsWith("@@")) {
      console.log(`\x1B[36m${line}\x1B[0m`);
    } else if (line.startsWith("-")) {
      console.log(`\x1B[31m${line}\x1B[0m`);
    } else if (line.startsWith("+")) {
      console.log(`\x1B[32m${line}\x1B[0m`);
    } else {
      console.log(line);
    }
  }
}
function resolveFilePath2(cwd, config, filePath) {
  const normalizedPath = filePath.replace(/\\/g, "/");
  if (normalizedPath.startsWith("/") || normalizedPath.includes("../")) {
    throw new Error(`Invalid registry file path: ${filePath}`);
  }
  const aliasPrefix = extractAliasPrefix2(config);
  if (filePath.startsWith("components/ui/")) {
    const uiAlias = config.aliases.ui || `${config.aliases.components}/ui`;
    const targetPath2 = path5.resolve(
      cwd,
      filePath.replace("components/ui/", stripAliasPrefix3(uiAlias, aliasPrefix) + "/")
    );
    return ensurePathInProjectRoot2(cwd, targetPath2, filePath);
  }
  if (filePath.startsWith("lib/")) {
    const libAlias = config.aliases.lib || `${aliasPrefix}lib`;
    const targetPath2 = path5.resolve(
      cwd,
      filePath.replace("lib/", stripAliasPrefix3(libAlias, aliasPrefix) + "/")
    );
    return ensurePathInProjectRoot2(cwd, targetPath2, filePath);
  }
  const targetPath = path5.resolve(cwd, filePath);
  return ensurePathInProjectRoot2(cwd, targetPath, filePath);
}
function ensurePathInProjectRoot2(cwd, targetPath, sourcePath) {
  const relative = path5.relative(cwd, targetPath);
  if (relative.startsWith("..") || path5.isAbsolute(relative)) {
    throw new Error(`Refusing to read outside project root for ${sourcePath}`);
  }
  return targetPath;
}
function extractAliasPrefix2(config) {
  const prefixes = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixes) {
    if (config.aliases.components.startsWith(prefix)) {
      return prefix;
    }
  }
  return "@/";
}
function stripAliasPrefix3(alias, prefix) {
  if (alias.startsWith(prefix)) {
    return alias.slice(prefix.length);
  }
  return alias;
}

// src/commands/update.ts
import path6 from "path";
import fs6 from "fs-extra";
import { Command as Command5 } from "commander";
import prompts3 from "prompts";
import { z as z6 } from "zod";
import * as Diff2 from "diff";
var updateOptionsSchema = z6.object({
  components: z6.array(z6.string()),
  cwd: z6.string(),
  all: z6.boolean(),
  force: z6.boolean(),
  dryRun: z6.boolean()
});
var update = new Command5().name("update").description("update components to the latest registry version").argument("[components...]", "the components to update").option(
  "-c, --cwd <cwd>",
  "the working directory. defaults to the current directory.",
  process.cwd()
).option("-a, --all", "update all installed components.", false).option("-f, --force", "skip confirmation prompt and overwrite.", false).option("--dry-run", "show what would be updated without making changes.", false).action(async (components, opts) => {
  try {
    const options = updateOptionsSchema.parse({
      components,
      cwd: path6.resolve(opts.cwd),
      all: opts.all,
      force: opts.force,
      dryRun: opts.dryRun
    });
    await runUpdate(options);
  } catch (error) {
    handleError(error);
  }
});
async function runUpdate(options) {
  const { components, cwd, all, force, dryRun } = options;
  const config = await getConfig(cwd);
  if (!config) {
    logger.error(
      `No ${highlighter.info("components.json")} file found at ${highlighter.info(cwd)}.`
    );
    logger.info("Run `npx neobrutal init` to initialize the project.");
    return;
  }
  let componentsToUpdate = components;
  if (all) {
    const checkSpinner = spinner("Scanning for installed components...").start();
    componentsToUpdate = await getInstalledComponents(cwd, config);
    checkSpinner.succeed(`Found ${componentsToUpdate.length} installed components.`);
  }
  if (componentsToUpdate.length === 0) {
    logger.warn("No components specified. Use --all to update all installed components.");
    return;
  }
  logger.break();
  logger.info(`Checking ${componentsToUpdate.length} component(s) for updates...`);
  logger.break();
  const updates = [];
  for (const componentName of componentsToUpdate) {
    try {
      const registryItem = await getRegistryItem(componentName);
      const componentUpdates = await checkComponentForUpdates(cwd, config, registryItem);
      if (componentUpdates.length > 0) {
        updates.push({
          name: componentName,
          files: componentUpdates
        });
      }
    } catch {
      logger.warn(`Component ${highlighter.warn(componentName)} not found in registry.`);
    }
  }
  if (updates.length === 0) {
    logger.success("All components are up to date!");
    return;
  }
  logger.info(`Found updates for ${updates.length} component(s):`);
  logger.break();
  for (const update2 of updates) {
    logger.log(`  ${highlighter.info(update2.name)}`);
    for (const file of update2.files) {
      const additions = countAdditions(file.localContent, file.registryContent);
      const deletions = countDeletions(file.localContent, file.registryContent);
      logger.log(`    ${file.path} (+${additions}/-${deletions})`);
    }
  }
  if (dryRun) {
    logger.break();
    logger.info("Dry run complete. No files were modified.");
    return;
  }
  logger.break();
  if (!force) {
    const { proceed } = await prompts3({
      type: "confirm",
      name: "proceed",
      message: "Do you want to apply these updates?",
      initial: true
    });
    if (!proceed) {
      logger.info("Update cancelled.");
      return;
    }
  }
  const updateSpinner = spinner("Applying updates...").start();
  let updatedCount = 0;
  for (const update2 of updates) {
    for (const file of update2.files) {
      const transformedContent = transformImports(file.registryContent, config);
      await fs6.writeFile(file.localPath, transformedContent, "utf-8");
      updatedCount++;
    }
  }
  updateSpinner.succeed(`Updated ${updatedCount} file(s).`);
  logger.break();
  logger.success("Update complete!");
}
async function getInstalledComponents(cwd, config) {
  const uiDir = resolveAliasPath(cwd, config.aliases.ui || `${config.aliases.components}/ui`);
  if (!await fs6.pathExists(uiDir)) {
    return [];
  }
  const index = await getRegistryIndex();
  const installedComponents = [];
  for (const item of index) {
    if (item.name === "utils") continue;
    const componentPath = path6.resolve(uiDir, `${item.name}.tsx`);
    if (await fs6.pathExists(componentPath)) {
      installedComponents.push(item.name);
    }
  }
  return installedComponents;
}
async function checkComponentForUpdates(cwd, config, registryItem) {
  const updates = [];
  for (const file of registryItem.files) {
    const localPath = resolveFilePath3(cwd, config, file.path);
    if (!await fs6.pathExists(localPath)) {
      continue;
    }
    const localContent = await fs6.readFile(localPath, "utf-8");
    const transformedRegistryContent = transformImports(file.content, config);
    if (localContent !== transformedRegistryContent) {
      updates.push({
        path: file.path,
        localPath,
        localContent,
        registryContent: file.content
      });
    }
  }
  return updates;
}
function countAdditions(localContent, registryContent) {
  const changes = Diff2.diffLines(localContent, registryContent);
  let count = 0;
  for (const change of changes) {
    if (change.added) {
      count += change.count || 1;
    }
  }
  return count;
}
function countDeletions(localContent, registryContent) {
  const changes = Diff2.diffLines(localContent, registryContent);
  let count = 0;
  for (const change of changes) {
    if (change.removed) {
      count += change.count || 1;
    }
  }
  return count;
}
function resolveFilePath3(cwd, config, filePath) {
  const normalizedPath = filePath.replace(/\\/g, "/");
  if (normalizedPath.startsWith("/") || normalizedPath.includes("../")) {
    throw new Error(`Invalid registry file path: ${filePath}`);
  }
  const aliasPrefix = extractAliasPrefix3(config);
  if (filePath.startsWith("components/ui/")) {
    const uiAlias = config.aliases.ui || `${config.aliases.components}/ui`;
    const targetPath2 = path6.resolve(
      cwd,
      filePath.replace("components/ui/", stripAliasPrefix4(uiAlias, aliasPrefix) + "/")
    );
    return ensurePathInProjectRoot3(cwd, targetPath2, filePath);
  }
  if (filePath.startsWith("lib/")) {
    const libAlias = config.aliases.lib || `${aliasPrefix}lib`;
    const targetPath2 = path6.resolve(
      cwd,
      filePath.replace("lib/", stripAliasPrefix4(libAlias, aliasPrefix) + "/")
    );
    return ensurePathInProjectRoot3(cwd, targetPath2, filePath);
  }
  const targetPath = path6.resolve(cwd, filePath);
  return ensurePathInProjectRoot3(cwd, targetPath, filePath);
}
function resolveAliasPath(cwd, alias) {
  const prefixes = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixes) {
    if (alias.startsWith(prefix)) {
      const targetPath2 = path6.resolve(cwd, alias.slice(prefix.length));
      return ensurePathInProjectRoot3(cwd, targetPath2, alias);
    }
  }
  const targetPath = path6.resolve(cwd, alias);
  return ensurePathInProjectRoot3(cwd, targetPath, alias);
}
function ensurePathInProjectRoot3(cwd, targetPath, sourcePath) {
  const relative = path6.relative(cwd, targetPath);
  if (relative.startsWith("..") || path6.isAbsolute(relative)) {
    throw new Error(`Refusing to access path outside project root for ${sourcePath}`);
  }
  return targetPath;
}
function extractAliasPrefix3(config) {
  const prefixes = ["@/", "~/", "#/", "$/"];
  for (const prefix of prefixes) {
    if (config.aliases.components.startsWith(prefix)) {
      return prefix;
    }
  }
  return "@/";
}
function stripAliasPrefix4(alias, prefix) {
  if (alias.startsWith(prefix)) {
    return alias.slice(prefix.length);
  }
  return alias;
}

// src/index.ts
var require2 = createRequire(import.meta.url);
var packageJson = require2("../package.json");
process.on("SIGINT", () => process.exit(0));
process.on("SIGTERM", () => process.exit(0));
async function main() {
  const program = new Command6().name("neobrutal").description("Add Neobrutal UI components to your project").version(
    packageJson.version,
    "-v, --version",
    "display the version number"
  );
  program.addCommand(init).addCommand(add).addCommand(list).addCommand(diff).addCommand(update);
  program.parse();
}
main();
